PsfLauncher/main.cpp

```diff
@@ -13,223 +13,227 @@
 #include <psf_runtime.h>
 #include <shellapi.h>
 #include <combaseapi.h>
+#include <ppltasks.h>
+#include <ShObjIdl.h>
+#include "ErrorInformation.h"
 
-using namespace std::literals;
+//These two macros don't exist in RS1.  Define them here to prevent build
+//failures when building for RS1.
+#ifndef PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE
+#    define PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE PROCESS_CREATION_DESKTOP_APPX_OVERRIDE
+#endif
 
-// Forward declaration
-void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t * executable, const wchar_t * arguments, bool wait, bool asadmin);
+#ifndef PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY
+#    define PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY PROC_THREAD_ATTRIBUTE_CHILD_PROCESS_POLICY
+#endif
 
-static inline bool check_suffix_if(iwstring_view str, iwstring_view suffix)
-{
-    if ((str.length() >= suffix.length()) && (str.substr(str.length() - suffix.length()) == suffix))
-    {
-        return true;
-    }
-
-    return false;
-}
+using namespace std::literals;
 
-void Log(const char* fmt, ...)
+struct ExecutionInformation
 {
-    std::string str;
-    str.resize(256);
-
-    va_list args;
-    va_start(args, fmt);
-    std::size_t count = std::vsnprintf(str.data(), str.size() + 1, fmt, args);
-    assert(count >= 0);
-    va_end(args);
-
-    if (count > str.size())
-    {
-        str.resize(count);
-
-        va_list args2;
-        va_start(args2, fmt);
-        count = std::vsnprintf(str.data(), str.size() + 1, fmt, args2);
-        assert(count >= 0);
-        va_end(args2);
-    }
+    LPCWSTR ApplicationName;
+    std::wstring CommandLine;
+    LPCWSTR CurrentDirectory;
+};
+
+//Forward declarations

```

Comment:
use noexcept where possible

Reply:
Why would I add noexcept?  From what I've read noexcept tells runtime that no exception will be thrown from this function and if there is one std::terminate is called instead.

Reply:
it also tells calling functions the same thing, so they in turn can omit catch handlers.  in other words, the runtime only drops in the std::terminate if you're calling functions that are not noexcept, which is suboptimal.  so it has a viral effect and you should get in the habit of applying it everywhere possible.

Reply:
Done.